Runtime/HistoryPanel.cs

```diff
@@ -0,0 +1,168 @@
+            await FetchHistoryAsync(searchParameters);
+        }
+
+        private void OnGeneratorStateChanged()
+        {
+            if (_runtimeGuiManager.Generator.CurrentState == BlockadeLabsSkyboxGenerator.State.Ready)
+            {
+                _stylePickerPanel.SetStyles(_runtimeGuiManager.Generator.AllModelStyleFamilies);
+            }
+        }
+
+        private async Task FetchHistoryAsync(HistorySearchQueryParameters searchParameters = null, bool clearResults = true)
+        {
+            if (_isFetchingHistory) { return; }
+            _isFetchingHistory = true;
+
+            try
+            {
+                if (clearResults)
+                {
+                    ClearHistory();
+                }
+
+                searchParameters ??= _lastQueryParams ??= new HistorySearchQueryParameters();
+                searchParameters.Limit = _pageSize;
+                _lastHistoryResult = await ApiRequests.GetSkyboxHistoryAsync(_lastQueryParams = searchParameters);
+
+                foreach (var item in _lastHistoryResult.data)
+                {
+                    var historyItemBehaviour = Instantiate(_historyItemPrefab, _historyItemsContainer);
+                    historyItemBehaviour.gameObject.SetActive(false);
+                    historyItemBehaviour.SetItemData(item, OnHistoryItemClick, OnHistoryItemDelete, OnHistoryItemDownload);
+                    _historyItems.Add(historyItemBehaviour);
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+            }
+            finally
+            {
+                _isFetchingHistory = false;
+            }
+        }
+
+        private void ClearHistory()
+        {
+            _lastQueryParams = null;
+
+            foreach (var historyItem in _historyItems)
+            {
+                Destroy(historyItem.gameObject);
+            }
+
+            _historyItems.Clear();
+        }
+
+        private void OnHistoryItemClick(ImagineResult imagineResult, Texture2D preview, Texture2D depth)
+            => _runtimeGuiManager.PreviewPopup.ShowPreviewPopup(imagineResult, preview, depth);
+
+        private void OnHistoryItemDelete(ImagineResult imagineResult)
+            => _runtimeGuiManager.DialogPopup.ShowDialog(
+                "Are you absolutely sure?",
+                "Are you sure you want to remove your skybox? This is not reversible.",
+                onConfirm: async () =>
+                {
+                    var result = await ApiRequests.DeleteSkyboxAsync(imagineResult.id);
+
+                    if (result)
+                    {
+                        await FetchHistoryAsync();
+                    }
+                },
+                onCancel: () =>
+                {
+                    // empty callback will close dialog
+                });
+
+        private async void OnHistoryItemDownload(ImagineResult imagineResult)
+        {
+#if UNITY_EDITOR
```

Comment:
why only editor?

Reply:
bc there may not be a place to download and use skybox at runtime (i.e. webgl)

only makes sense to view it.